src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1748,6 +1752,21 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                         ImmutableArray.Create(new TypedConstant(compilation.GetWellKnownType(WellKnownType.System_Type), TypedConstantKind.Type, originalType)),
                         isOptionalUse: true));
             }
+
+            if (this.IsMicrosoftCodeAnalysisEmbeddedAttribute() && GetEarlyDecodedWellKnownAttributeData() is null or { HasCodeAnalysisEmbeddedAttribute: false })
+            {
+                // This is Microsoft.CodeAnalysis.EmbeddedAttribute, and the user didn't manually apply this attribute to itself. Grab the parameterless constructor
```

Comment:
>// This is Microsoft.CodeAnalysis.EmbeddedAttribute [](http://example.com/codeflow?start=16&length=51)

Consider adding an assert to verify the fact that "This is Microsoft.CodeAnalysis.EmbeddedAttribute"

Reply:
The `if` condition verifies that. I'm not sure what an additional assert that just does the same logic adds?

Reply:
You are right.